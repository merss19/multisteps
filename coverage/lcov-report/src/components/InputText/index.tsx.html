<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/InputText/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/InputText</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">69.7% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/25</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react'
import {WrappedFieldProps} from 'redux-form'
import * as classNames from 'classNames'
<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ></span></span></span></span>
interfac<span class="fstat-no" title="function not covered" ><span class="branch-3 cbranch-no" title="branch not covered" >e </span>OwnProps extends<span class="cstat-no" title="statement not covered" > WrappedFieldProp<span class="cstat-no" title="statement not covered" >s&lt;{}&gt; {<span class="cstat-no" title="statement not covered" ></span></span></span></span>
	placeholder: string
}
&nbsp;
class InputText extends React.PureC<span class="branch-0 cbranch-no" title="branch not covered" >omponent&lt;OwnProps</span>, {}&gt; {
&nbsp;
	render() {
		const {input, placeholder, meta: {touched, error}} = this<span class="fstat-no" title="function not covered" ><span class="branch-3 cbranch-no" title="branch not covered" >.p</span>rops</span>
		re<span class="cstat-no" title="statement not covered" >turn (</span>
			&lt;div <span class="cstat-no" title="statement not covered" >className={classNa</span>mes('input', {'input--error': touched &amp;&amp; error})}&gt;
				&lt;inp<span class="cstat-no" title="statement not covered" >ut<span class="cstat-no" title="statement not covered" ></span></span>
					classNa<span class="cstat-no" title="statement not covered" >me='input__fi</span>eld'
					{...input}
<span class="cstat-no" title="statement not covered" >					type='tex</span>t'
					placeholder={placeholder}
				/&gt;
				{touched &amp;&amp; error &amp;&amp; &lt;p className='input__alert'&gt;{error}&lt;/p&gt;}
			&lt;/div&gt;
		)
	}
}
export default InputText</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 27 2017 14:35:22 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
