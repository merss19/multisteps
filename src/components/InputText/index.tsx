import * as React from 'react'
import {WrappedFieldProps} from 'redux-form'
import * as classNames from 'classNames'

interface OwnProps extends WrappedFieldProps<{}> {
	placeholder: string
}

class InputText extends React.PureComponent<OwnProps, {}> {

	render() {
		const {input, placeholder, meta: {touched, error}} = this.props
		return (
			<div className={classNames('input', {'input--error': touched && !!error})}>
				<input
					className='input__field'
					{...input}
					type='text'
					placeholder={placeholder}
				/>
				{touched && error && <p className='input__alert'>{error}</p>}
			</div>
		)
	}
}
export default InputText