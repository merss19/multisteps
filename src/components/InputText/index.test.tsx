import {shallow} from 'enzyme'
import InputText from './index'
import * as React from 'react'
import toJson from 'enzyme-to-json'

let touched = false
let props = {
	placeholder: 'Ваш email',
	meta: {
		touched: touched,
		error: 'Email должен быть заполнен',
		asyncValidating: false,
		autofilled: false,
		invalid: false,
		pristine: false,
		dirty: false,
		dispatch: (fn: any) => fn,
		valid: false
	},

}

describe('Component InputText', () => {
	const component = shallow(<InputText {...props}/>)
	it('renders component', () => {
		expect(component).toHaveLength(1)
	})
	it('renders correctly', () => {
		const tree = toJson(shallow(<InputText {...props}/>))
		expect(tree).toMatchSnapshot()
	})
	it('render placeholder', () => {
		expect(component.find('.input__field').prop('placeholder')).toEqual('Ваш email')
	})
	it('render input', () => {
		const component = shallow(<InputText {...props}/>)
		expect(component.find('.input__field').exists()).toBe(true)
	})
	describe('Render error', () => {
		beforeEach(() => {
			props.meta.touched = true
		})
		it('render error', () => {
			const component = shallow(<InputText {...props}/>)
			expect(component.find('.input__alert').exists()).toBe(true)
		})
		it('render error text', () => {
			const component = shallow(<InputText {...props}/>)
			expect(component.find('.input__alert').text()).toEqual('Email должен быть заполнен')
		})
		it('render error class', () => {
			const component = shallow(<InputText {...props}/>)
			expect(component.find('.input--error').exists()).toBe(true)
		})
	})
	describe('do not Render error', () => {
		beforeEach(() => {
			props.meta.touched = true
			props.meta.error = ''
		})
		it('do not render error', () => {
			const component = shallow(<InputText {...props}/>)
			expect(component.find('.input__alert').exists()).toBe(false)
		})
		it('do not render error class', () => {
			const component = shallow(<InputText {...props}/>)
			expect(component.find('.input--error').exists()).toBe(false)
		})
	})
})
