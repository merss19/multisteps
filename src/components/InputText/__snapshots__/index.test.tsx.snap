// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InputText renders correctly 1`] = `
<div
  className="input"
>
  <input
    className="input__field"
    placeholder="Ваш email"
    type="text"
  />
</div>
`;
